src/testing/io.zig

```diff
@@ -0,0 +1,209 @@
+const std = @import("std");
+const os = std.os;
+const posix = std.posix;
+const mem = std.mem;
+const assert = std.debug.assert;
+const log = std.log.scoped(.io);
+
+const stdx = @import("../stdx.zig");
+const constants = @import("../constants.zig");
+const FIFO = @import("../fifo.zig").FIFO;
+const Time = @import("../time.zig").Time;
+const buffer_limit = @import("../io.zig").buffer_limit;
+const DirectIO = @import("../io.zig").DirectIO;
+
+/// A very simple mock IO implementation that only implements what is needed to test Storage.
+pub const IO = struct {
+    pub const fd_t = u32;
+
+    /// Options for fault injection during fuzz testing.
+    pub const Options = struct {
+        /// Seed for the storage PRNG.
+        seed: u64 = 0,
+
+        /// Chance out of 100 that any read will return an error.InputOutput.
+        read_fault_probability: u8 = 0,
+
+        /// Chance out of 100 that a read larger than a logical sector
+        /// will return an error.InputOutput.
+        larger_than_logical_sector_read_fault_probability: u8 = 0,
+    };
+
+    files: []const []u8,
+    options: Options,
+    prng: std.rand.DefaultPrng,
+
+    completed: FIFO(Completion) = .{ .name = "io_completed" },
+
+    pub fn init(files: []const []u8, options: Options) IO {
```

Comment:
Can we assert files.len > 0?

Reply:
Having zero files is perfectly fine and supported, although it would be weird. I'll add an assertion in `read` and `write` to ensure that the `fd_t` is within the length of `files`, though.